src/Server/Workspace.php

```diff
@@ -200,4 +214,44 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * @param Options|null $settings
+     */
+    public function didChangeConfiguration(Options $settings = null)
+    {
+        if ($settings === null) {
+            return;
+        }
+
+        $changedOptions = $this->getChangedOptions($settings);
+        $this->options = $settings;
+
+        if (!empty(array_intersect($changedOptions, $this->options->getIndexerOptions()))) {
+            // check list of options that changed since last time against the list of valid indexer options
+
+            // start wiping from the main index
+            $this->index->wipe();
+
+            // check for existing indexer and start indexing
+            if ($this->indexer) {
+                $this->indexer->index()->otherwise('\\LanguageServer\\crash');
+            }
+        }
+    }
+
+    /**
+     * Get a list with all options that changed since last time
+     *
+     * @param Options $settings
+     * @return array List with changed options
+     */
+    private function getChangedOptions(Options $settings): array
+    {
+        // squash nested array for comparing changed options
+        $old = array_map('json_encode', get_object_vars($this->options));
+        $new = array_map('json_encode', get_object_vars($settings));
```

Comment:
JSON encoding is not a good way to do deep comparison. PHP can do this natively with `==`.